Assets/Mirror/Runtime/Logging/NetworkLogSettings.cs

```diff
@@ -0,0 +1,45 @@
+using UnityEngine;
+
+namespace Mirror.Logging
+{
+    /// <summary>
+    /// Used to load LogSettings in build
+    /// </summary>
+    [DisallowMultipleComponent]
+    [AddComponentMenu("Network/NetworkLogSettings")]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkLogSettings.html")]
+    public class NetworkLogSettings : MonoBehaviour
+    {
+        [SerializeField] internal LogSettings settings;
+
+#if UNITY_EDITOR
+        // called when component is added to GameObject
+        void Reset()
+        {
+            LogSettings existingSettings = EditorLogSettingsLoader.FindLogSettings();
+            if (existingSettings != null)
+            {
+                settings = existingSettings;
+
+                UnityEditor.EditorUtility.SetDirty(this);
```

Comment:
why do you use SetDirty here?
Undo.RecordObject is the new way to do this, SetDirty doesn't always work the way you would expect it to work. We use Undo in sceneId assignment too.

btw if you use Undo, you need to put the RecordObject call **before** modify settings (it records the changes)

Reply:
I can't find much documentation on which one to use, but from my experience `SetDirty` works fine for Assets (ScriptableObject/Prefabs).

Reply:
Just tested it, SetDirty still seems to work, even with multiple scenes loaded.

Pressing reset on the component only marks the scene it is in as dirty.

![image](https://user-images.githubusercontent.com/23101891/81415786-cc50c000-9140-11ea-816f-139a6caccfe1.png)

Reply:
It seems that the Reset button is what marks the scene as dirty. However if NetworkManager is a prefab then we need this SetDirty to mark the prefab as dirty.

Reply:
yes, i believe that was the issue with setdirty when I tried it before.
Undo sets scene as dirty properly.
but if setdirty works here then that's ok